lib/logstash/inputs/elasticsearch.rb

```diff
@@ -264,6 +253,11 @@ class LogStash::Inputs::Elasticsearch < LogStash::Inputs::Base
   # If set, the _source of each hit will be added nested under the target instead of at the top-level
   config :target, :validate => :field_reference
 
+  # Obsolete Settings
+  config :ssl, :validate => :boolean, :default => false, :obsolete => "Set 'ssl_enabled' instead."
```

Comment:
Thinking about this, we can probably skip the `:validate` and `:default` parts, and stick with `:obsolete` to avoid users fixing up settings, only to find they were obsolete anyway. WDYT?

(My only caveat is to keep `:password` and `:uri` (not applicable here), as they are considered secret, and are [obfuscated](https://github.com/elastic/logstash/blob/a94659cf82d7e62c5a5f550df3e4e5bd0c6cf3e1/logstash-core/lib/logstash/config/mixin.rb#L608-L615) on discovery)

Reply:
Nice, yeah i got tripped up on this writing tests. I was surprised to find that `:validate` would trigger when the option is `:obsolete`. I may be misunderstanding the test scenario but it looks to me like as a user I may be frustrated with a case whereby i set the wrong thing, then get a validation error that its the wrong form, only to be rewarded with a "this is no longer a valid option" once I satisfy the input validation. Not a huge deal but certainly a surprise if that is the case.